tests/test_textcat.py

```diff
@@ -0,0 +1,96 @@
+from pathlib import Path
+from typing import Callable, List
+
+import pytest
+import spacy
+
+from recipes.openai import OpenAISuggester, get_api_credentials, load_template
+from recipes.openai_textcat import DEFAULT_PROMPT_PATH, make_textcat_response_parser
+
+
+def make_suggester(
+    response_parser: Callable,
+    labels: List[str],
+    prompt_path: Path,
+    model: str = "text-davinci-003",
+    **kwargs
+) -> OpenAISuggester:
+    if "openai_api_key" not in kwargs or "openai_api_org" not in kwargs:
+        api_key, api_org = get_api_credentials(model)
+        if "openai_api_key" not in kwargs:
+            kwargs["openai_api_key"] = api_key
+        if "openai_api_org" not in kwargs:
+            kwargs["openai_api_org"] = api_org
+    if "max_examples" not in kwargs:
+        kwargs["max_examples"] = 0
+    if "prompt_template" not in kwargs:
+        kwargs["prompt_template"] = load_template(prompt_path)
+    if "segment" not in kwargs:
+        kwargs["segment"] = False
+    if "openai_model" not in kwargs:
+        kwargs["openai_model"] = "text-davinci-003"
+    return OpenAISuggester(response_parser=response_parser, labels=labels, **kwargs)
+
+
+@pytest.mark.parametrize(
+    # fmt: off
+    "response,answer",
+    [
+        ("Answer: accept\nReason: The text is a recipe.", True),
+        ("Answer: Accept\nReason: The text is a recipe.", True),
+        ("Answer: reject\nReason: The text is not a recipe.", False),
+        ("Answer: Reject\nReason: The text is not a recipe.", False),
+        ("answer: reject\nreason: The text is not a recipe.", False),
+        ("answer: Reject\nreason: The text is not a recipe.\nI don't know what it's about.", False),
+    ],
+    # fmt: on
+)
+def test_parse_response_binary(response, answer):
+    """Test if parse response works for common examples"""
+    labels = ["recipe"]  # binary
+    parser = make_textcat_response_parser(labels=labels)
+    example = parser(response)
+    assert example.get("answer") == answer
+
+
+@pytest.mark.parametrize(
+    # fmt: off
+    "response,answer",
+    [
+        ("Answer: recipe,feedback,question\nReason: It is all three!", ["recipe", "feedback", "question"]),
+        ("Answer: recipe\nReason: The text is a recipe.", ["recipe"]),
+        ("Answer: recipe,\nReason: The text is a recipe.", ["recipe"]),
+        ("Answer: recipe,feedback,\nReason: The text is a feedback about a recipe.\nThat's what I think...", ["recipe", "feedback"]),
+        ("answer: recipe,feedback,\nreason: The text is a feedback about a recipe.\nThat's what I think...", ["recipe", "feedback"]),
+        ("answer: \nreason: It's none of the above.", []),
```

Comment:
I wonder ... here's some annotated data with `textcat.manual`. 

```bash
python -m prodigy textcat.manual go-emo doubt.jsonl --label excitement
```

I've removed some keys for brevity. 

```
{"text":"OMG THOSE TINY SHOES!!! *desire to boop snoot intensifies*","label":"excitement","_view_id":"classification","answer":"accept"}
{"text":"Wow dan I like being up at 4am....","label":"excitement","_view_id":"classification","answer":"reject"}
```

Notice how the answer is `reject` when the class in question is not found? It seems that your parser does not do that. 

```python
> parser = make_textcat_response_parser(labels=["excitement"])
> parser("Answer: \nReason: No class detected.")
{'answer': False,
 'label': 'excitement',
 'meta': {'answer': '', 'reason': 'No class detected.'}}
```

Shouldn't the `"answer"` key be `accept` or `reject` here? Just to be consistent with Prodigy's output.

Reply:
Got it. I might have based it from a textcat project that uses boolean for the keys. Will update that.

Reply:
Ah this is solved!